### PR TITLE
selection-stable indentation + multi-line piece decorations + some regrouting bug fixes

### PR DESCRIPTION
**Pull request by dm0n3y**

### CODE REVIEW

File: src/core/lang/Form.re [hunk 1]
```diff
@@ -77,7 +77,7 @@ let forms: list((string, t)) = [
   ("array_access", mk(ii, ["[", "]"], mk_post(P.ap, Exp, [Exp]))),
   ("parens_exp", mk(ii, ["(", ")"], mk_op(Exp, [Exp]))),
   ("parens_pat", mk(ii, ["(", ")"], mk_op(Pat, [Pat]))),
-  ("fun_", mk(di, ["fun", "=>"], mk_pre(P.fun_, Pat, [Exp]))),
+  ("fun_", mk(di, ["fun", "=>"], mk_pre(P.fun_, Exp, [Pat]))),
```
Comment (dm0n3y) [inline src/core/lang/Form.re:80]: also this includes a form fix

File: src/core/tiles/Segment.re [hunk 2]
```diff
@@ -113,28 +113,54 @@ let remold = (seg: t): list(t) =>
...
+      | Pre(_, r) => go(snd(root_mold.nibs).sort, r)
+      | Post(l, _) => go(fst(root_mold.nibs).sort, l)
+      | Bin(l, _, r) =>
+        go(fst(root_mold.nibs).sort, l) + go(snd(root_mold.nibs).sort, r)
+      };
+    r + unichild_rs;
+  };
+
+  go(s, skel(seg));
+};
```
Comment (dm0n3y) [inline src/core/tiles/Segment.re:156]: also includes sort ranking that will likely need debugging

File: src/web/view/Code.re [hunk 3]
```diff
@@ -5,35 +5,37 @@ open Util;
...
-         | Piece.Whitespace({content: c, _}) when c == Whitespace.linebreak => [
+    |> List.map((p: Piece.t) => {
+         let m = Measured.find_p(p, M.map);
+         switch (p) {
+         | Piece.Whitespace(w) when w.content == Whitespace.linebreak => [
              span_c("whitespace", [text(Whitespace.linebreak)]),
              Node.br([]),
              Node.text(
-               String.concat("", List.init(indent, _ => Unicode.nbsp)),
+               String.concat("", List.init(m.last.col, _ => Unicode.nbsp)),
```
Comment (dm0n3y) [inline src/web/view/Code.re:18]: `Code.Text` no longer needs to duplicate indent logic, just looks up indentation on each whitespace in the measured map

**Comment by disconcision** [comment]
whoops i didnt even intentionally merge this? probably fine tho?

**Pull request merged**